### PR TITLE
Joint Servo CBM

### PR DESCRIPTION
**Pull request by nexusmrsep**
#### Summary
```SUMMARY: Features "Joint Servo CBM - movement enhancer CBM"```

#### Purpose of change
Idea from Discourse Forum. CBM that acts as a "reverse" Joint Torsion Ratchet CBM
Servomotors installed in leg joints provide power-assisted movement, thus faster movement.

#### Describe the solution
It adds a new movement-enhancing CBM.
Its features:
EDITED
- when active enhances **running** speed by **15%** at cost of **20 energy** per movement 
- when active enhances **walking** speed by **5%** at the cost of **10** energy per movement
- when **passive** it slows movement by **10%** as its connected directly to leg joints, and it creates some resistance/tension
- reduces stamina cost of using muscle powered vehicles (bike for ex.) at the cost of power

Also added Bionic Runner profession, so you can use this from the beginning (stacks nicely with fleet footed, etc.)

#### Describe alternatives you've considered
- not adding the CBM
- inventing some other interactions/combos

#### Additional context
N/A

### CODE REVIEW

File: src/monattack.cpp [hunk 1]
```diff
@@ -1194,8 +1194,9 @@ bool mattack::science( monster *const z ) // I said SCIENCE again!
 
             // if the player can see it
             if( g->u.sees( *z ) ) {
-                add_msg( m_warning, _( "The %s shudders, and some sort of caustic fluid leaks from a its damaged shell!" ),
-                z->name().c_str() );
+                add_msg( m_warning,
```
Comment (anothersimulacrum) [inline src/monattack.cpp]: This looks unrelated and is a change made in #28569. Is this intentional?

Comment (nexusmrsep) [reply src/monattack.cpp]: Astyle fail in Gorgon forced me to astyle this bit.

**Comment by kevingranade** [comment]
Fleet plus this CBM is way OP, slashing move cost to 56% of normal.  Even when walking you'll be able to out-speed most anything while on flat ground, and I don't think there's anything that can keep up with you running.

I'd prefer to drop the speedup by a bit and maybe also increase the power consumption.  Other options include only triggering the speedup when running so you're at least limited by stamina or possibly adding an "overheat" mechanic to the CBM to prevent continuous use.

The mechanism is a bit iffy, but that's not that big a deal, I think it just needs to be described as a bit more invasive than it currently is.

**Comment by Night-Pryanik** [comment]
The situation is worsened by the fact of existence of Synaptic Accelerator CBM, giving you a flat bonus to speed.

**Comment by nexusmrsep** [comment]
How about equalizing the bonus/malus at +10% speed active and -10% speed passive? Plus upping power consumption x2 to 10 power units per square?

**Comment by Hirmuolio** [comment]
How about having a sort of stacking penalties system for bonuses?

Idea from EVE online. Multiple % bonuses to same stat get weaker the more you stack them.
https://wiki.eveuniversity.org/Stacking_penalties

So for example if you have 3x +10% bonuses you would only get 10%, 8.9% and 5.71% bonuses applied on you.

The bonuses could then stay strong on their own without worrying about how OP they are when they are stacked.

**Comment by nexusmrsep** [comment]
> How about having a sort of stacking penalties system for bonuses?

While a diminishing rate of return sounds good, it'd have to be a more throughout change that would be out of the scope.

**Comment by nexusmrsep** [comment]
I've re-balanced the CBM according to ideas provided, and updated the PR description. I was tempted with overheat mechanic, but adding a new variable just to store the usage of a CBM didn't seem right to me.

**Comment by int-ua** [comment]
Is it hard to change power consumption based on weight of PC with items?

**Comment by nexusmrsep** [comment]
I guess I'd be possible with a few checks, but how would that work in your idea?

**Comment by int-ua** [comment]
Force = mass × acceleration, so to accelerate more mass you need a greater force. Am I missing something? Maybe it's not strictly 1:1 because of some factors, but in general you need more energy to move more mass.

File: data/json/bionics.json [hunk 2]
```diff
@@ -1038,6 +1038,14 @@
     "occupied_bodyparts": [ [ "ARM_L", 8 ], [ "ARM_R", 8 ], [ "LEG_L", 12 ], [ "LEG_R", 12 ] ],
     "flags": [ "BIONIC_POWER_SOURCE", "BIONIC_NPC_USABLE", "BIONIC_TOGGLED" ]
   },
+  {
+    "id": "bio_jointservo",
+    "type": "bionic",
+    "name": "Joint Servo",
+    "description": "Your leg joints have been equipped with servomotors that provide power-assisted movement. They are optimized for running, but walking also requires less effort when this bionic is online. However, when it's offline it will hamper your movement, as you struggle against it's moving parts.",
```
Comment (Night-Pryanik) [inline data/json/bionics.json]:
```suggestion
    "description": "Your leg joints have been equipped with servomotors that provide power-assisted movement.  They are optimized for running, but walking also requires less effort when this bionic is online.  However, when it's offline it will hamper your movement, as you struggle against its moving parts.",
```

File: data/json/items/bionics.json [hunk 3]
```diff
@@ -824,6 +824,15 @@
     "price": 380000,
     "difficulty": 4
   },
+  {
+    "id": "bio_jointservo",
+    "copy-from": "bionic_general",
+    "type": "BIONIC_ITEM",
+    "name": "Joint Servo CBM",
+    "description": "Your leg joints have been equipped with servomotors that provide power-assisted movement. They are optimized for running, but walking also requires less effort when this bionic is online. However, when it's offline it will hamper your movement, as you struggle against it's moving parts.",
```
Comment (Night-Pryanik) [inline data/json/items/bionics.json]:
```suggestion
    "description": "Your leg joints have been equipped with servomotors that provide power-assisted movement.  They are optimized for running, but walking also requires less effort when this bionic is online.  However, when it's offline it will hamper your movement, as you struggle against its moving parts.",
```

File: data/json/professions.json [hunk 4]
```diff
@@ -1288,6 +1288,19 @@
       "female": [ "sports_bra", "panties" ]
     }
   },
+  {
+    "type": "profession",
+    "ident": "bionic_runner",
+    "name": "Bionic Runner",
+    "description": "You were that kind of sportsman who couldn't get off the track. You love running, and you enhanced your body to do it even better. Now there is plenty to run from, but this is your kind of game.",
```
Comment (Night-Pryanik) [inline data/json/professions.json]:
```suggestion
    "description": "You were that kind of sportsman who couldn't get off the track.  You love running, and you enhanced your body to do it even better.  Now there is plenty to run from, but this is your kind of game.",
```

File: src/game.cpp [hunk 5]
```diff
@@ -11311,6 +11311,14 @@ void game::on_move_effects()
             u.charge_power( 1 );
         }
     }
+    if( u.has_active_bionic( bionic_id( "bio_jointservo" ) ) ) {
+        if( u.move_mode == "run" ) {
+            u.charge_power( -20 );
+        } else {
+            u.charge_power( -10 );            
+        }
+
```
Comment (Night-Pryanik) [inline src/game.cpp]: Empty line.

**Comment by nexusmrsep** [comment]
jenkins rebuild

**Comment by int-ua** [comment]
Found the video I thought about the moment I saw this PR: https://www.youtube.com/watch?v=N28bttqLoZE Multi-joint, personalized soft exosuit by team from Harvard
